File: app/src/main/java/com/github/premnirmal/ticker/network/historicaldata/History.java
package com.github.premnirmal.ticker.network.historicaldata;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;

/**
 * Created by premnirmal on 12/30/14.
 */
public class History implements Parcelable {

    public List<Quote> quote;

    public History() {

    }

    public History(Parcel in) {
        in.readList(quote, Quote.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<History> CREATOR = new Creator<History>() {
        public History createFromParcel(Parcel in) {
            return new History(in);
        }

        public History[] newArray(int size) {
            return new History[size];
        }
    };

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeList(quote);
    }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/network/data/historicaldata/History.kt
package com.github.premnirmal.ticker.network.data.historicaldata

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

/**
 * Created by premnirmal on 3/30/17.
 */
class History : Parcelable {

  @SerializedName("quote") var quote: List<Quote>? = null

  constructor()

  constructor(`in`: Parcel) {
    `in`.readList(quote, Quote::class.java.classLoader)
  }

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeList(quote)
  }

  companion object {

    val CREATOR: Creator<History> = object : Creator<History> {
      override fun createFromParcel(`in`: Parcel): History {
        return History(`in`)
      }

      override fun newArray(size: Int): Array<History> {
        return newArray(size)
      }
    }
  }
}